persistence/src/main/java/de/cyface/persistence/model/GeoLocation.kt

```diff
@@ -0,0 +1,170 @@
+     * pole) and 90.0 (north pole).
+     * @param lon The captured longitude of this data point in decimal coordinates as a value between -180.0
+     * and 180.0.
+     * @param altitude The captured altitude of this data point in meters above WGS 84.
+     * @param speed The current speed of the measuring device according to its location sensor in meters per second.
+     * @param accuracy The current accuracy of the measuring device in meters.
+     * @param verticalAccuracy The current vertical accuracy of the measuring device in meters.
+     * @param measurementId The device-unique id of the measurement this data point belongs to.
+     */
+    constructor(
+        timestamp: Long, lat: Double, lon: Double, altitude: Double?,
+        speed: Double, accuracy: Double?, verticalAccuracy: Double?, measurementId: Long
+    ) : this(
+        0,
+        timestamp,
+        lat,
+        lon,
+        altitude,
+        speed,
+        accuracy,
+        verticalAccuracy,
+        null,
+        measurementId
+    )
+
+    /**
+     * Creates a new instance of this class which was not yet persisted and has [id] set to `0`.
+     *
+     * @param location The cached [ParcelableGeoLocation] to create the [GeoLocation] from.
+     * @param measurementId The device-unique id of the measurement this data point belongs to.
+     */
+    constructor(location: ParcelableGeoLocation, measurementId: Long) : this(
+        location.timestamp, location.lat, location.lon, location.altitude,
+        location.speed, location.accuracy, location.verticalAccuracy, measurementId
+    )
+
+    // is timestamp checked?
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other == null || javaClass != other.javaClass) return false
+        val that = other as GeoLocation
+        return (that.lat.compareTo(lat) == 0 && that.lon.compareTo(lon) == 0
+                && that.speed.compareTo(speed) == 0 && that.accuracy == accuracy
+                && measurementId == that.measurementId && altitude == that.altitude
+                && verticalAccuracy == that.verticalAccuracy && isValid == that.isValid)
+    }
+
+    // FIXME: I changed the hashCode to `id` only like in all other @Entity classes,
+    // as the constructor which has `id` set to `0` is usually only used to insert an entry into the db.
+    // The only issue which could occur would be if we want to insert a set or new points into the database,
+    // as the set would then just have one of these points as entry, as `id` is `0`.
+    // But for Measurement, GeoLocation, Event and Pressure we usually insert one point at a time.
+    // DEPRECATED: To ease migration with `main` branch, we keep the models similar to `GeoLocation` but might want to change this
+    // DEPRECATED: in future. https://github.com/cyface-de/android-backend/pull/258#discussion_r1070618174
+    override fun hashCode(): Int {
```

Comment:
In our [previous discussion](https://github.com/cyface-de/android-backend/pull/258#discussion_r1070618174) about `hashCode()` we postponed changing `hashCode()` until we are on the main branch again, which we are now.

As I reimplemented the models with Room's `@Entity` I wanted to set `hashCode()` to just compare the `id` as you suggested some time back.

But when we insert new entity into the database we use the constructor which sets `id` to `0`. This constructor is only used to insert a new entry into the db. When loading entries from the db, the id is set correctly by Room.

I.e. just comparing id in hashCode would be okay, unless we decide at some point in the future to insert a `Set<>` of new points into the database, as the `Set` would then just contain one entry (as `id` is `0`). Right now we only insert one Measurement, GeoLocation, Event and Pressure at a time.